src/Core/Rendering/RenderController.cpp

```diff
@@ -537,6 +538,62 @@ namespace MxEngine
         this->RenderToTexture(output, shader);
     }
 
+    void RenderController::ApplyGodRayEffect(CameraUnit& camera, TextureHandle& input, TextureHandle& output)
+    {
+        if (!camera.Effects->IsGodRayEnabled())
+            return;
+        int dirLightCount = this->Pipeline.Lighting.DirectionalLights.size();
+        if (dirLightCount <= 0)
+            return;
+        auto godRayShader = this->Pipeline.Environment.Shaders["GodRay"_id];
+        godRayShader->Bind();
+        godRayShader->IgnoreNonExistingUniform("camera.viewProjMatrix");
+        godRayShader->IgnoreNonExistingUniform("normalTex");
+        godRayShader->IgnoreNonExistingUniform("albedoTex");
+        godRayShader->IgnoreNonExistingUniform("materialTex");
+
+        Texture::TextureBindId textureId = 0;
+        this->BindGBuffer(camera, *godRayShader, textureId);
+        this->BindCameraInformation(camera, *godRayShader);
+
+        input->Bind(textureId++);
+        godRayShader->SetUniform("cameraOutput", input->GetBoundId());
+
+        {
+            constexpr size_t MaxDirLightCount = 4;
+            const auto& dirLights = this->Pipeline.Lighting.DirectionalLights;
+            size_t lightCount = Min(MaxDirLightCount, dirLights.size());
+
+            godRayShader->SetUniform("lightCount", (int)lightCount);
+            godRayShader->SetUniform("rayColor", camera.Effects->GetGodRayColor());
+
+            for (size_t i = 0; i < lightCount; i++)
+            {
+                auto& dirLight = this->Pipeline.Lighting.DirectionalLights[i];
+
+                Vector4 colorPacked = Vector4(dirLight.Color * dirLight.Intensity, dirLight.AmbientIntensity);
```

Comment:
lets not dublicate we code (see the same parameters bound in dir lighting pass), I suggest creating a helper function which binds dir light info to shader